ultravox/model/ultravox_model.py

```diff
@@ -237,20 +237,28 @@ def prepare_inputs_for_generation(
         past_key_values: Optional[Union[Tuple, transformers.cache_utils.Cache]] = None,
         attention_mask: Optional[torch.Tensor] = None,
         inputs_embeds: Optional[torch.Tensor] = None,
+        cache_position: Optional[torch.Tensor] = None,
         **kwargs,
     ) -> Dict[str, Any]:
         model_input = self.language_model.prepare_inputs_for_generation(
             input_ids=input_ids,
             past_key_values=past_key_values,
             attention_mask=attention_mask,
             inputs_embeds=inputs_embeds,
+            cache_position=cache_position,
             **kwargs,
         )
 
```

Comment:
How did this work before, if we were not offsetting the audio_token_start_idx by the prefill_start_idx?

Reply:
Previously we use `is_cache_empty(past_key_values)` to determine if we need to set the `audio` fields. The `audio` fields are only needed for the prefilling stage, which is the first generation step and `past_key_values=None`.